source/common/event/dispatcher_impl.cc

```diff
@@ -41,6 +41,14 @@ DispatcherImpl::DispatcherImpl(Buffer::WatermarkFactoryPtr&& factory, Api::Api&
 
 DispatcherImpl::~DispatcherImpl() {}
 
+void DispatcherImpl::initializeStats(Stats::Scope& parent_scope) {
+  std::string thread_id = api_.threadFactory().currentThreadId()->debugString();
```

Comment:
@eziskind has two additional requests that I think I should put some work into:
* The main thread should have an obvious name, not just the lowest-numbered thread ID.
* It might also be useful to aggregate the stats for the worker threads, since that's probably what most users will want to see. This would be in addition to the per-thread stats.

Reply:
I would argue that (1) is useful, i.e. main thread, but (2) can/should be done by a monitoring system and we don't need to have redundant stats. That said, I don't feel that strongly about (2), so if others reckon it's a good idea, sure, why not.

Reply:
Added (1), punted on (2) after confirming that @eziskind is ok with it.